dask/dataframe/io/csv.py

```diff
@@ -620,6 +620,9 @@ def read_pandas(
     # Use sample to infer dtypes and check for presence of include_path_column
     head_kwargs = kwargs.copy()
     head_kwargs.pop("skipfooter", None)
+    if head_kwargs.get("engine") == "pyarrow":
+        # Use c engine to infer since Arrow engine does not support nrows
+        head_kwargs["engine"] = "c"
```

Comment:
Should we set this to `None` instead to use whatever the default in `pandas` is? Probably not a huge deal either way, just wondering if there's any downside to hardcoding `"c"`

Reply:
We could pop the key, None wouldn't work. I don't think that we would ever change it in pandas, but better be save

Reply:
Changed my mind, we might switch to arrow at some point, better to keep c here